src/tests/tracing/runtimeeventsource/RuntimeEventSourceTest.cs

```diff
@@ -1,114 +1,101 @@
-
-                    // Wait for events.
-                    Thread.Sleep(1000);
-
-                    // Generate some GC events.
-                    GC.Collect(2, GCCollectionMode.Forced);
-
-                    // Wait for more events.
-                    Thread.Sleep(1000);
-
-                    // Ensure that we've seen some events.
-                    Assert.True("listener.EventCount > 0", listener.EventCount > 0);
-                }
+            Console.WriteLine("Event: " + k + " " + v);
+        }
+        return 1;
+    }
+}
 
-                // Generate some more GC events.
-                GC.Collect(2, GCCollectionMode.Forced);
+public class RuntimeEventListener : EventListener
+{
+    internal readonly Dictionary<string, bool> ObservedEvents = new Dictionary<string, bool>();
 
-                // Ensure that we've seen no events.
-                Assert.True("noEventsListener.EventCount == 0", noEventsListener.EventCount == 0);
-            }
+    private static readonly string[] s_expectedEvents = new[] {
+        "appContextSwitch",
+        "appContextBoolAsStringData",
+        "RuntimeHostConfigSwitch", // Set in the project file
+    };
 
-            return 100;
-        }
+    private static readonly string[] s_unexpectedEvents = new[] {
+        "appContextBoolData",
+        "appContextStringData",
+    };
 
-        private static void Allocator()
+    protected override void OnEventSourceCreated(EventSource source)
+    {
+        if (source.Name.Equals("System.Runtime"))
         {
-            while (true)
-            {
-                for(int i=0; i<1000; i++)
-                    GC.KeepAlive(new object());
-
-                Thread.Sleep(10);
-            }
+            EnableEvents(source, EventLevel.Informational, (EventKeywords)1 /* RuntimeEventSource.Keywords.AppContext */);
         }
     }
 
-    internal sealed class SimpleEventListener : EventListener
+    protected override void OnEventWritten(EventWrittenEventArgs eventData)
     {
-        private string m_name;
-
-        // Keep track of the set of keywords to be enabled.
-        public static EventKeywords EnableKeywords
-        {
-            get;
-            set;
-        }
-
-        public SimpleEventListener(string name)
+        // Check AppContext switches
+        if (eventData is { EventName: "LogAppContextSwitch",
+                           Payload: { Count: 2 } })
         {
-            m_name = name;
+            var switchName = (string)eventData.Payload[0];
+            ObservedEvents[switchName] = ((int)eventData.Payload[1]) == 1;
+            return;
         }
+    }
 
-        public int EventCount { get; private set; } = 0;
-
-        protected override void OnEventSourceCreated(EventSource eventSource)
+    public bool Verify()
+    {
+        foreach (var key in s_expectedEvents)
         {
-            if (eventSource.Name.Equals("Microsoft-Windows-DotNETRuntime"))
+            if (!ObservedEvents[key])
```

Comment:
Maybe have separate checks and failure messages not having switch logged at all versus having the switch logged with the wrong value?

Reply:
Not opposed, but I'm hoping to take this for RC 1, so I'd prefer to make test improvements in a later iteration